### DISCUSSION TITLE
Dockge fails to start due to missing file in ./data

### DISCUSSION THREAD

**Discussion by Katherine1** [louislam/dockge#672] [Ask for Help]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### 📝 Describe your problem

Followed installation steps.
1. Created /opt/dockge and /opt/stacks as root
2. Downloaded compose.yaml to /opt/dockge
3. Started dockge with docker-compose up -d

Rather than run as expected, Dockge creates the ./data directory, but then errors about a missing config .json in that directory. It looks like Dockge creates the data directory, but does not populate it so it fails when trying to open a file in that directory.

### 📝 Error Message(s) or Log

2024-11-18T01:32:07Z [SERVER] INFO: Welcome to dockge!
2024-11-18T01:32:07Z [SERVER] INFO: NODE_ENV: production
2024-11-18T01:32:07Z [SERVER] INFO: Server Type: HTTP
2024-11-18T01:32:07Z [SERVER] INFO: Data Dir: ./data/
2024-11-18T01:32:07Z [DB] WARN: ENOENT: no such file or directory, open 'data/db-config.json'
2024-11-18T01:32:07Z [SERVER] ERROR: Failed to prepare your database: EACCES: permission denied, open 'data/db-config.json'

### 🐻 Dockge Version

1.4.2

### 💻 Operating System and Arch

Fedora 40 x86

### 🌐 Browser

LibreWolf 132.0.2-1

### 🐋 Docker Version

Podman 5.2.3

**Comment by mizady** [comment]
this sounds like it could be a permissions issue, have you tried a different directory or using chmod to verify that permissions are not an issue?  I do this by running ls -l  in the parent directory and look at the current permissions, then chmod 777 /opt/dockge/data or even perhaps /opt/dockge  and try spinning it up again.  it is not a good idea to leave the folder at 777 for security reasons so if this resolves the issue then you may need to chown the dockge directory to root or whatever user id that docker is attached to (default is root)

**Reply by Katherine1** [reply]
chmod 777 -R  on the data directory didn't affect anything. Note that docker-compose was able to create the data directory itself just fine.

**Reply by mizady** [reply]
i tried to recreate the issue and sadly no luck. I've had this issue before and I don't recall what the cause was.  have you tried changing the version tag from version 1? i use the latest tag when playing with dockge vanilla version so not sure if that has an impact. The command listed above does not include sudo, have you tried it using sudo docker compose up -d  ? (also note the lack of - as well as that is technically a different executable that is now deprecated)

**Reply by Katherine1** [reply]
As I'm on Fedora, I have podman, not docker. I am currently using the version 1 tag. I'll try latest.

**Reply by mizady** [reply]
yea i noticed the os and podman after i replied.  I hadn't worked with podman and its been a while for fedora so i may not get this one.  I ran into the same problem on ubuntu server a few months back and i wiped it and used my modified version of dockge but then when i went to do it again it worked so i'm not 100% sure what I changed to fix it.  If you want to try it to see if it'll work you're welcome to, it's based one the repository as of a few months ago but that was when node was still v18 not updated as it was recently.  once the container buttons has been merged i'll be switching back to regular dockge.  thejericko/dockge:latest is the tag for it on docker hub.

**Reply by Katherine1** [reply]
The version 1 tag and latest were last updated 10 months ago and seem to have the same build. The nightly tag was updated 1 month ago, so I tried that. No change.

**Reply by mizady** [reply]
alright hopefully someone with some podman experience will jump in here.  I would try launching it as a podman command and not as a compose file to see if that works as well. perhaps --privileged maybe necessary

**Comment by hatruongm** [comment]
I'm in the same boat, I love Dockge. Look like no one care about Fedora. I'm waiting for awhile. I love Fedora more than others.

**Comment by jquagga** [comment]
It's a fedora/selinux thing.  You need to add :Z to the data line to add the selinux context.  Then it should come up normally (or you could use a podman volume instead of directory I'd imagine).  

```
- ./data:/app/data:Z
```

**Reply by ojdo** [reply]
Fedora 42 user here (w/ native `docker-compose`), I needed to add `:Z` as well to make DB initialization work. Does someone have an insight as to why this is needed? The data directory does not seem to be anything special. Docker docs on `:z`/`:Z` options for volume mounts: [Configure the SELinux label](https://docs.docker.com/engine/storage/bind-mounts/#configure-the-selinux-label)
